File: app/src/test/java/br/eng/ecarrara/vilibra/fixture/BookFixture.java
package br.eng.ecarrara.vilibra.fixture;

import java.util.Arrays;
import java.util.Calendar;
import java.util.Date;

import br.eng.ecarrara.vilibra.domain.entity.Book;

import static br.eng.ecarrara.vilibra.data.VilibraContract.getDateFromDb;
import static br.eng.ecarrara.vilibra.data.VilibraContract.getDbDateString;

public class BookFixture {

    /**
     * Create a test domain Book with some fake data.
     * "Devs Test Book"
     */
    public static Book getTestBookDevsTestBook() {
        // This is done to make sure that we have a compatible format with the one Vilibra uses.
        Date publishingDate = getDateFromDb(getDbDateString(Calendar.getInstance().getTime()));

        return new Book.Builder(1L, "Devs Test Book")
                .setSubtitle("A for devs to use tests to test the code with too much tests.")
                .setIsbn13("1234567890123")
                .setIsbn10("1234567890")
                .setPageCount(434)
                .setAuthors(Arrays.asList("Crazy Dev Yo", "Man of Code"))
                .setPublishedDate(publishingDate)
                .setPublisher("Herbert Richards")
                .build();
    }

    /**
     * Create a test domain Book with some fake data.
     * "Dominando Android"
     */
    public static Book getTestBookDominandoAndroid() {
        // This is done to make sure that we have a compatible format with the one Vilibra uses.
        Date publishingDate = getDateFromDb("20140101");

        return new Book.Builder(1L, "Dominando o Android")
                .setSubtitle("Do básico ao avançado")
                .setIsbn13("9788575224120")
                .setIsbn10("8575224123")
                .setPageCount(792)
                .setAuthors(Arrays.asList("Nelson Glauber"))
                .setPublishedDate(publishingDate)
                .setPublisher("Editora Novatec")
                .build();
    }

    /**
     * Create a test domain Book with some fake data.
     * "Pro Android 4"
     */
    public static Book getTestBookProAndroid4() {
        // This is done to make sure that we have a compatible format with the one Vilibra uses.
        Date publishingDate = getDateFromDb("20120101");

        return new Book.Builder(1L, "Pro Android 4")
                .setSubtitle("")
                .setIsbn13("9781430239314")
                .setIsbn10("143023931X")
                .setPageCount(1020)
                .setAuthors(Arrays.asList("Satya Komatineti", "Dave MacLean"))
                .setPublishedDate(publishingDate)
                .setPublisher("Apress")
                .build();
    }


}

File: app/src/test/kotlin/br/eng/ecarrara/vilibra/fixture/BookFixture.kt
package br.eng.ecarrara.vilibra.fixture

import br.eng.ecarrara.vilibra.book.domain.entity.Book
import br.eng.ecarrara.vilibra.core.data.datasource.contentprovider.VilibraContract.getDateFromDb
import br.eng.ecarrara.vilibra.core.data.datasource.contentprovider.VilibraContract.getDbDateString
import java.util.*

object BookFixture {

    /**
     * Create a test domain Book with some fake data.
     * "Devs Test Book"
     */
    // This is done to make sure that we have a compatible format with the one Vilibra uses.
    val testBookDevsTestBook: Book
        get() {
            val publishingDate = getDateFromDb(getDbDateString(Calendar.getInstance().time))

            return Book(
                    id = 1L,
                    title = "Devs Test Book",
                    subtitle = "A for devs to use tests to test the code with too much tests.",
                    isbn13 = "1234567890123",
                    isbn10 = "1234567890",
                    pageCount = 434,
                    authors = Arrays.asList("Crazy Dev Yo", "Man of Code"),
                    publishedDate = publishingDate,
                    publisher = "Herbert Richards"
            )
        }

    /**
     * Create a test domain Book with some fake data.
     * "Dominando Android"
     */
    // This is done to make sure that we have a compatible format with the one Vilibra uses.
    val testBookDominandoAndroid: Book
        get() {
            val publishingDate = getDateFromDb("20140101")

            return Book(
                    id = 1L,
                    title = "Dominando o Android",
                    subtitle = "Do básico ao avançado",
                    isbn13 = "9788575224120",
                    isbn10 = "8575224123",
                    pageCount = 792,
                    authors = Arrays.asList("Nelson Glauber"),
                    publishedDate = publishingDate,
                    publisher = "Editora Novatec"
            )

        }

    /**
     * Create a test domain Book with some fake data.
     * "Pro Android 4"
     */
    // This is done to make sure that we have a compatible format with the one Vilibra uses.
    val testBookProAndroid4: Book
        get() {
            val publishingDate = getDateFromDb("20120101")

            return Book(
                    id = 1L,
                    title = "Pro Android 4",
                    isbn13 = "9781430239314",
                    isbn10 = "143023931X",
                    pageCount = 1020,
                    authors = Arrays.asList("Satya Komatineti", "Dave MacLean"),
                    publishedDate = publishingDate,
                    publisher = "Apress"
            )

        }

}
